### ISSUE TITLE
Using Code component throws error with vite in production

### ISSUE THREAD

**Issue by ayZagen** [expressive-code/expressive-code#264]
The following messages printed while building:
```
Ambiguous external namespace resolution: "node_modules/.pnpm/expressive-code@0.38.0/node_modules/expressive-code/dist/index.js" re-exports "loadShikiTheme" from one of the external modules "@expressive-code/plugin-frames", "@expressive-code/plugin-shiki" and "@expressive-code/plugin-text-markers", guessing "@expressive-code/plugin-frames".

"ExpressiveCodeTheme", "loadShikiTheme", "ExpressiveCodeBlock", "getStableObjectHash" and "stableStringify" are imported from external module "@expressive-code/plugin-text-markers" but never used in "ec.config.mjs" and "node_modules/.pnpm/expressive-code@0.38.0/node_modules/expressive-code/dist/index.js".

"toText" is imported from external module "hast-util-to-text" but never used in "node_modules/.pnpm/@expressive-code+core@0.38.0/node_modules/@expressive-code/core/dist/hast.js" and "node_modules/.pnpm/@expressive-code+core@0.38.0/node_modules/@expressive-code/core/dist/index.js".

"visitParents", "CONTINUE", "EXIT" and "SKIP" are imported from external module "unist-util-visit-parents" but never used in "node_modules/.pnpm/@expressive-code+core@0.38.0/node_modules/@expressive-code/core/dist/hast.js" and "node_modules/.pnpm/@expressive-code+core@0.38.0/node_modules/@expressive-code/core/dist/index.js".

"s" is imported from external module "hastscript" but never used in "node_modules/.pnpm/@expressive-code+core@0.38.0/node_modules/@expressive-code/core/dist/hast.js" and "node_modules/.pnpm/@expressive-code+core@0.38.0/node_modules/@expressive-code/core/dist/index.js".

"ExpressiveCodeTheme", "ExpressiveCodeBlock", "getStableObjectHash" and "stableStringify" are imported from external module "@expressive-code/plugin-frames" but never used in "node_modules/.pnpm/expressive-code@0.38.0/node_modules/expressive-code/dist/index.js".

```

And following error is thrown in runtime:
```
There was an uncaught error in the middle of the stream while rendering /page. Error: Failed to render a `<Code>` component on page http://localhost:4321/page:

The requested module '@expressive-code/plugin-frames' does not provide an export named 'loadShikiTheme'
    at file:///<loc>/dist/server/chunks/MDCode_Dz-TKLPm.mjs:130:11 {
  [cause]: file:///<loc>/dist/server/chunks/index_DZwu4QU4.mjs:4
  import { pluginFrames as pluginFrames$1, loadShikiTheme } from '@expressive-code/plugin-frames';
                                           ^^^^^^^^^^^^^^
  SyntaxError: The requested module '@expressive-code/plugin-frames' does not provide an export named 'loadShikiTheme'
      at ModuleJob._instantiate (node:internal/modules/esm/module_job:177:21)
      at async ModuleJob.run (node:internal/modules/esm/module_job:260:5)
      at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:543:26)
      at async createRenderer (file:///<loc>/dist/server/chunks/MDCode_Dz-TKLPm.mjs:54:57)
}

```

`ec.config.mjs`:
```js
import { pluginShiki } from "@expressive-code/plugin-shiki";
import { pluginTextMarkers } from "@expressive-code/plugin-text-markers";
import { pluginCollapsibleSections } from '@expressive-code/plugin-collapsible-sections'
import { pluginLineNumbers } from '@expressive-code/plugin-line-numbers'

/** @type {import('astro-expressive-code').AstroExpressiveCodeOptions} */
export default {
  plugins: [
    pluginShiki(),
    pluginLineNumbers(),
    pluginCollapsibleSections(),
    pluginTextMarkers()
  ],
  themes: ['github-dark', 'github-light'],
  themeCssRoot: `html`,
  themeCssSelector: (theme) => `[data-code-theme='${theme.name}']`,
  useDarkModeMediaQuery: false,
  defaultProps: {
    showLineNumbers: false,
    frame: 'none'
  },
  frames: {
    showCopyToClipboardButton: true,
    removeCommentsWhenCopyingTerminalFrames: true
  },
  styleOverrides: {
    frames: {
      editorTabBorderRadius: '0',
      shadowColor: 'transparent',
    },
    borderRadius: '0',
  },
}
```

**Comment by hippotastic** [comment]
Can you please provide a minimal reproduction for this on StackBlitz? I cannot reproduce this locally and the EC docs homepage builds fine as well.

**Comment by hippotastic** [comment]
Oh, I think I see it! You're adding all the bundled default plugins again in your `plugins` option in the EC config file. Please try removing them. :)

**Comment by ayZagen** [comment]
hi @hippotastic thanks for the quick reply, removing them from config file didn't work but after removing them as dependencies the error is gone. 

However, after using it like that shiki and text markers seems not loaded. Note, that they are working in dev but not after build.

and here is the usage of `Code` component:
```astro
---
import {Code} from "astro-expressive-code/components"

const {ins, del, mark, content, lang, ...otherProps} = Astro.props
const meta = `ins={${ins || ''}} del={${del || ''}} ${mark || ''}`
---

<Code
  {...otherProps}
  code={content}
  meta={meta}
  lang={Astro.props['data-language'] || lang}
/>

```

**Comment by hippotastic** [comment]
Hmm, I can't tell why this happens in your project, I really think a StackBlitz repro would help a lot!

You can see on my EC docs homepage that both Shiki and all annotations like text markers work fine, and it's also built with Astro. It must be something special about your setup.

**Comment by ayZagen** [comment]
Here is the [reproduction](https://stackblitz.com/edit/github-8z9kpg) (prebuilt, and automatically runs the built server). If you run in dev it works just ok.

**Comment by hippotastic** [comment]
Thank you, that was really helpful!

It seems that the Vite instance responsible for rendering the `Code` component has trouble resolving the `@expressive-code/core` module that the plugins depend on. This only becomes apparent when using `pnpm` due to its more strict dependency handling.

You can fix the build output by doing `pnpm add @expressive-code/core`, which seems to make the core module available to the plugins as well. I have to admit that I don't understand yet why this is happening and how I can solve it, but at least this should make the site work again for you.

What confuses me even more is that this issue does not appear in my EC monorepo which also uses `pnpm`. I assume that module resolution logic in a monorepo is different and therefore this issue never happened in my own projects.

I'll keep investigating this to find a true fix.

**Comment by ayZagen** [comment]
Thanks for the detailed insights. adding `@expressive-code/core` as direct dependency indeed solved the problem.
I thought the error was happening because of ssr build as astro injects custom logic to build process and asset resolving. I haven't tried but maybe the issue won't exists with static build. 

Thanks again for your work. I leave closing issue to you, you may want to track the process of it.

**Comment by hippotastic** [comment]
The error seems to be independent from the build type - I could also reproduce it in a static build.

Happy to hear that the workaround solved the problem for you. I'll keep this issue here open until I've released the fix to NPM shortly.

**Comment by hippotastic** [comment]
This is now fixed in version 0.38.1, which I just released to NPM. Thank you for the report and reproduction!
